metabase/metabase
# Model Actions: Create Actions in Models

## Description

This simply allows us to associate newly-created actions with models. It adds a model picker form component and uses that component in the action save modal.

It does not
- have the slick UI from figma
- allow moving actions between models
- allow model creation from the action save flow
(yet)

## Screenshots
![saveActionOnModel](https://user-images.githubusercontent.com/30528226/192563540-3449fc87-a7a3-4fb2-8c42-4d81abfa0d24.gif)

The update form removes the model selection option (for now at least)

![Screen Shot 2022-09-27 at 8 54 14 AM](https://user-images.githubusercontent.com/30528226/192563593-f204ee18-f0a2-4669-9261-9e8714f4a415.png)

## Testing Steps

- create a new action and select a model to save

How do you know an action got saved on a model? Check the POST to `http://localhost:3000/api/model-action` you should see the `card_id` for the model you selected.

![Screen Shot 2022-09-27 at 9 04 21 AM](https://user-images.githubusercontent.com/30528226/192563956-49b73715-a7bb-43ad-b749-51a56d7619fa.png)

- update an existing action, and ensure you can save it, but there should be no API activity for the model-aciton endpoint.
